### DISCUSSION TITLE
Automatically move SPAM to Junk folder

### DISCUSSION THREAD

**Discussion by alexanderadam** [docker-mailserver/docker-mailserver#3141] [Q&A]
Somehow I wasn't able to find it in the documentation.
Maybe I'm just not good in searching or does it still need to be documented?

**Comment by polarathene** [comment]
Docs can probably be improved.

There's a FAQ entry that mentions two ENV that do this for you: https://docker-mailserver.github.io/docker-mailserver/v11.3/faq/#how-can-i-make-spamassassin-better-recognize-spam

Those ENV should be enabled by default:
- https://docker-mailserver.github.io/docker-mailserver/v11.3/config/environment/#spamassassin_spam_to_inbox
- https://docker-mailserver.github.io/docker-mailserver/v11.3/config/environment/#move_spam_to_junk

You may need to enable some spam related services if that's not working:
- [`ENABLE_SPAMASSASSIN`](https://docker-mailserver.github.io/docker-mailserver/v11.3/config/environment/#enable_spamassassin) (_disabled by default_).
- [`ENABLE_AMAVIS`](https://docker-mailserver.github.io/docker-mailserver/v11.3/config/environment/#enable_amavis) (_enabled by default_).

Once mail is properly detected as spam, it would then be delivered to your junk folder. That should be handled by the two earlier ENV mentioned which uses a Dovecot sieve config to make it all happen:

https://github.com/docker-mailserver/docker-mailserver/blob/f0edcc28d9ddfdae8fa95260c71dab25c53b419f/target/scripts/startup/setup.d/dovecot.sh#L77-L85

**Reply by alexanderadam** [reply]
Thank you so much for your response.

I see. I ignored the SpamAssassin variables since I'm using RSpamD.
I should've mentioned that before.

This setting is probably specific to SpamAssassin and the config for RSpamD doesn't have this feature yet, right?

**Reply by polarathene** [reply]
@georglauterbach would be best suited to answer anything about rspamd support.

You'll find some recent issues (now closed I think) discussed rspamd [like this one](https://github.com/docker-mailserver/docker-mailserver/issues/3138).

AFAIK the `SPAMASSASSIN_SPAM_TO_INBOX` doesn't depend on SpamAssassin to be running, the code snippet above is for Dovecot, and uses this sieve script:

https://github.com/docker-mailserver/docker-mailserver/blob/8ec5dbec74839e0460252bcc2115035310f49327/target/dovecot/sieve/before/60-spam.sieve#L1-L4

So if rspamd is marking mail as spam in the same way, it should work. @georglauterbach may want to rename the ENV before v12 release (_or deprecate the existing ENV until a later release_ :man_shrugging: ).

**Reply by georglauterbach** [reply]
This will actually not work and needs to be adjusted anyways: <https://workaround.org/ispmail/stretch/filtering-out-spam-with-rspamd> (due a slightly different header). I will add this to my backlog.

**Reply by alexanderadam** [reply]
@georglauterbach [thank you so much for considering this.](https://github.com/docker-mailserver/docker-mailserver/issues/3145)

**Reply by georglauterbach** [reply]
Tracking issues for this are #3145 and #3156
